modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/utils/rulesSupport.js

```diff
@@ -373,11 +373,22 @@ export const isActionsValid = (actions) =>
 				case 'auto-fill': {
 					const {inputs, outputs} = payload;
 
-					return (
-						Boolean(target) &&
-						(Boolean(Object.values(inputs).length) ||
-							Boolean(Object.values(outputs).length))
+					let valid =
+						Boolean(Object.values(inputs).length) ||
+						Boolean(Object.values(outputs).length);
+
+					const currentDataProvider = dataProviders.filter(
+						(provider) => provider.id === payload.label
 					);
+
+					if (currentDataProvider[0]?.inputOutputLength) {
+						valid =
+							Object.keys(outputs).length +
+								Object.values(inputs).length ===
```

Comment:
That's the second time the `outputs` and the `inputs` are been counted.

I suggest to extract this sum to a constant:
```javascript
const parametersQuantity = Object.keys(outputs).length + Object.values(inputs).length;
```
```suggestion
							parametersQuantity ===
```